src/Raygun.Diagnostics/RaygunTraceListener.cs

```diff
@@ -132,6 +133,9 @@ public override void TraceEvent(TraceEventCache eventCache, string source, Trace
     /// </PermissionSet>
     public override void TraceEvent(TraceEventCache eventCache, string source, TraceEventType eventType, int id, string format, params object[] args)
     {
+      var stackTrace = new StackTrace(true).GetFrame(1);
+      if (format == string.Empty) format = $"An unexpected error occurred while calling {stackTrace.GetMethod().Name} in {Path.GetFileName(stackTrace.GetFileName())} at line {stackTrace.GetFileLineNumber()}";
```

Comment:
Just one minor adjustment. Lets allow `null` as well. so `if (!string.IsNullOrEmpty(format))`.  

so this syntax would be valid 
```c#
Trace.TraceError(null, new { id = "1234", name = "Anon"});
```